test/libsolidity/semanticTests/abiencodedecode/abi_encode_with_selector.sol

```diff
@@ -0,0 +1,24 @@
+pragma abicoder v1;
+contract C {
+	function f0() public pure returns (bytes memory) {
+		return abi.encodeWithSelector(0x12345678);
+	}
+	function f1() public pure returns (bytes memory) {
+		return abi.encodeWithSelector(0x12345678, "abc");
+	}
+	function f2() public pure returns (bytes memory) {
+		bytes4 x = 0x12345678;
+		return abi.encodeWithSelector(x, "abc");
+	}
+	function f3() public pure returns (bytes memory) {
+		bytes4 x = 0x12345678;
+		return abi.encodeWithSelector(x, type(uint).max);
+	}
+}
+// ====
+// compileViaYul: also
+// ----
+// f0() -> 0x20, 4, 8234104107246695022420661102507966550300666591269321702959126607540084801536
+// f1() -> 0x20, 0x64, 8234104107246695022420661102507966550300666591269321702959126607540084801536, 862718293348820473429344482784628181556388621521298319395315527974912, 91135606241822717681769169345594720818313984248279388438121731325952, 0
+// f2() -> 0x20, 0x64, 8234104107246695022420661102507966550300666591269321702959126607540084801536, 862718293348820473429344482784628181556388621521298319395315527974912, 91135606241822717681769169345594720818313984248279388438121731325952, 0
+// f3() -> 0x20, 0x24, 0x12345678ffffffffffffffffffffffffffffffffffffffffffffffffffffffff, -26959946667150639794667015087019630673637144422540572481103610249216
```

Comment:
Is the result for `f3()` actually correct? Tests are passing so I guess it must be but according to my calculations `26959946667150639794667015087019630673637144422540572481103610249216` is `0x00000000010000000000000000000000000000000000000000000000000000000` (`2**224`).

Why not `0xffffffffff0000000000000000000000000000000000000000000000000000000`? I'd expect the tail of `u256(-1)` to spill over into the next slot but apparently it just gets cut off.

BTW, these expectations would be much easier to compare with the .cpp test if you kept the hex form (though I guess you'd have to copy it by hand instead of letting isoltest fill it).

Reply:
Yes, we really should update islotest to use hex for large numbers. Don't we actually use hex if it has more zeros?

I think the number is like that because everything is shifted by 4 bytes (the selector), so it does not align with the 32 byte "grid".

Reply:
Ah, looks like it's my mistake after all. The negative is 2's complement.

`2**256 - 26959946667150639794667015087019630673637144422540572481103610249216` is `0xffffffff00000000000000000000000000000000000000000000000000000000`.

Reply:
> Don't we actually use hex if it has more zeros?

Not for negatives: #9601. We had someone who volunteered to fix it but he's still missing in action :)

> I think the number is like that because everything is shifted by 4 bytes (the selector), so it does not align with the 32 byte "grid".

Yeah, that's why the `0x12345678ffffffffffffffff...` looked fine to me but I was puzzled by `0x00000000010000...` in the other slot. A shift should still keep `0xff` in most bytes. But looks like I just incorrectly converted the negative int to 256-bit hex. That's actually one more reason to fix the formatting in isoltest :)